### PR TITLE
Hotfix: PHP Fatal on inc/template-functions.php line 99

### PR DESCRIPTION
**Pull request by MiguelAxcar**
This PR intends to add a hotfix which was breaking role pages on `develop`.

Example of role page:
https://wikimediafoundation-org-develop.go-vip.co/role/board/

Error being thrown:
```
PHP message: PHP Fatal error: Uncaught TypeError: Illegal offset type in /var/www/wp-content/themes/shiro/inc/template-functions.php:99
```

I already deployed this hotfix to Shiro's `develop`, and updated WMF `develop`, fixing the issue, assuring that the fix was effective.

### CODE REVIEW

File: inc/template-functions.php
Comment (kadamwhite) [inline inc/template-functions.php:93]: If we're adding a type hint here we should also add the `int` hint to `wmf_get_posts_by_child_roles()`, so that we catch any type errors as early as possible.

File: inc/template-functions.php
Comment (kadamwhite) [inline inc/template-functions.php:106]: I think we can actually drop this comment, it describes the line of code exactly which feels duplicative. I don't believe this logic would be confusing without the comment.

**Review by kadamwhite — CHANGES_REQUESTED** [review]

File: inc/template-functions.php
Comment (kadamwhite) [inline inc/template-functions.php:108]: Ignore needs to be on the line above

**Review by kadamwhite — APPROVED** [review]
Approving because the code looks good, but the checker is still reporting an error. Once that is passing you can merge it, but please don't merge it before we get that little green checkmark

File: inc/template-functions.php
Comment (MiguelAxcar) [reply inc/template-functions.php:108]: I added this error log to have some info about what is exactly causing the issue, as a way to uncover the root cause.

**Pull request merged**
